app/models/manageiq/providers/google/cloud_manager/event_catcher/event_producer.rb

```diff
@@ -0,0 +1,62 @@
+# Helper class responsible for polling a Google Pubsub topic and retreiving
+# event type messages. Parsing of the message is deferred to the caller of
+# #poll.
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::EventProducer
+  def initialize(ems)
+    @ems = ems
+  end
+
+  # Poll for messages. This makes a service call to the Google Pubsub service
+  # and returns any messages generated from the activity log. Note that these
+  # messages are acknowledged before returning, so the caller has no
+  # responsibility to explicitly acknowledge them.
+  #
+  # @return [Array<Hash>] possibly empty list of messages
+  def poll
+    create_subscription unless subscription_exists?
+    # For now, return immediately with up to 10 messages
+    @ems.with_provider_connection(:service => 'pubsub') do |google|
+      subscription = google.subscriptions.get(subscription_name)
+
+      msgs = subscription.pull(:return_immediately => true, :max_messages => 10)
+      return [] if msgs.empty?
+
+      subscription.acknowledge(msgs)
+
+      decoded_msgs = msgs.map do |msg|
+        JSON.parse(msg.message['data'])
+      end
+
+      # Filter out messages not destined for us (we should only be consuming
+      # event log entries)
+      decoded_msgs.reject! do |msg|
+        msg['log'] != 'compute.googleapis.com/activity_log'
```

Comment:
If you're referring to the filtering out of activity log; not really. Anyone with permissions can publish any type of data into the pubsub queue.